inbm/dispatcher-agent/tests/unit/source/test_ubuntu_source_cmd.py

```diff
@@ -279,6 +279,31 @@ def test_successfully_remove_gpg_key_and_source_list(
         except SourceError:
             self.fail("Remove GPG key raised DispatcherException unexpectedly!")
 
+     @patch("dispatcher.packagemanager.package_manager.verify_source", side_effect=DispatcherException('error'))
+     def test_failed_add_gpg_key_method(self, mock_verify_source):
+        parameters = ApplicationAddSourceParameters(
+            gpg_key_uri="https://dl-ssl.google.com/linux/linux_signing_key.pub",
+            gpg_key_name="name"
+        )
+        command = UbuntuApplicationSourceManager()
+        try:
+            command.add(parameters)
+        except SourceError:
+            self.fail("Source Gpg key URI verification check failed: error")
+
+
+     @patch("dispatcher.packagemanager.package_manager.verify_source")
+     def test_success_add_gpg_key_method(self, mock_verify_source):
+        parameters = ApplicationAddSourceParameters(
+            gpg_key_uri="https://dl-ssl.google.com/linux/linux_signing_key.pub",
+            gpg_key_name="name"
+        )
+        command = UbuntuApplicationSourceManager()
+        try:
+            command.add(parameters)
+        except SourceError:
+             assert False, f"'UbuntuApplicationSourceManager.add' raised an exception {err}"
```

Comment:
In the `test_success_add_gpg_key_method`, use the `with self.assertRaises(SourceError):` context manager to properly test for exceptions. [important]